packages/keystone/ipBlackList.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express')
+const { isEmpty } = require('lodash')
+
+const conf = require('@open-condo/config')
+const { getIp } = require('@open-condo/keystone/ip.utils')
+
+const ipBlackList = (conf.IP_BLACK_LIST || '').split(',')
+    .map(ip => ip.trim())
+    .filter(ip => !isEmpty(ip))
+    .reduce((map, ip) => {
+        map[ip] = true
+        return map
+    }, {})
+
+
+class KeystoneIpBlackListMiddleware {
+    async prepareMiddleware ({ keystone }) {
+        // nosemgrep: javascript.express.security.audit.express-check-csurf-middleware-usage.express-check-csurf-middleware-usage
+        const app = express()
+        app.use((req, res, next) => {
+            const ip = getIp(req)
+            if (ipBlackList[ip]) {
+                res.status(429).send('Too Many Requests')
```

Comment:
It's better to use [410](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410)